packages/discovery-react-components/src/components/StructuredQuery/utils/formatSatisfyRulesDropdownMessage.tsx

```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+
+export const formatSatisfyRulesDropdownMessage = (message: string, dropdown: JSX.Element) => {
```

Comment:
i think this will be a useful utility across our components that we can make a bit more generic

something like

```suggestion
export const formatSatisfyRulesDropdownMessage = (message: string, dropdown: JSX.Element) => {

interface SubstitutionValues {
  [name: string]: React.Node // this includes strings as well as JSX elements i think
}

const SPLIT_VARIABLES_REGEX = /({[^}]+})/;
formatMessage(message: string, values: SubstitutionValues) => {
  const variables = Object.keys(values);
  return message.split(SPLIT_VARIABLES_REGEX).map((part, i) => {
   const variableWithoutBraces = part.replace(/{}/g, '');
   return part.contains('{') ? <div key={i}>{values[variableWithoutBraces]}</div> : part;
  });
}
```